sim_instr/src/sim_instr_node.cpp

```diff
@@ -65,27 +63,25 @@ int main(int argc, char** argv) {
     std::vector<ros::Timer> timers;
 	timers.reserve(lines.size());
 
-	bool lines_sorted;
-	for(int i = 0; i < (lines.size() - 1); ++i) {
-		if(!(lines.at(i + 1).time > lines.at(i).time)) {
-			ROS_WARN_STREAM("Commands in the Input file should be in time-increasing order, resolved internally");
-			lines_sorted = false;
-			break;
-		}
-	}
-
 
 	while (!ros::Time::isValid()) {
-		std::cout << "spin once" << std::endl;
 		ros::spinOnce();
 		
 	}
 
-	for (const TimeAndCmdVel &directive : lines) {
-		timers.push_back(nh.createTimer(directive.time, [cmd_vel = directive.cmd_vel, &pub](const ros::TimerEvent&) {
+	std::transform(lines.cbegin(), lines.cend(), std::back_inserter(timers), [&nh, &pub](const TimeAndCmdVel &directive) {
+		
+		auto callback = [cmd_vel = directive.cmd_vel, &pub](const ros::TimerEvent& event) {
 			pub.publish(cmd_vel);			
-		}, true));
-	}
+		};
+        return nh.createTimer(directive.time, callback, true);
+    });
+
 
+	if(!std::is_sorted(timers.begin(), timers.end())) {
```

Comment:
I looked into this because I was a little confused as to why this works. The implementation of ros::Timer is just a pointer to some opaque type and when you compare two Timer objects, you're actually just comparing those pointers. So there's a total ordering, but probably not the one you're looking for. Checking to see if the lines themselves are sorted, perhaps using this lambda:

```c++
[](const TimeAndCmdVel &lhs, const TimeAndCmdVel &rhs) { return lhs.time < rhs.time; }
```/